pkg/agent/config/config.go

```diff
@@ -679,3 +686,20 @@ func validateNetworkConfig(nodeConfig *config.Node) error {
 
 	return nil
 }
+
+// processFlannelOpts separates the three expected flannel options in three different variables
+func processFlannelOpts(opts []string) (bool, bool, error) {
+	var ipv6Masq, externalIP bool
+	for _, opt := range opts {
+		switch {
+		case strings.Contains(opt, "ipv6-masq"):
+			ipv6Masq = true
+		case strings.Contains(opt, "external-ip"):
+			externalIP = true
+		default:
+			return false, false, fmt.Errorf("wrong arg: %s passed in --flannel-opts. Correct args are backend, ipv6-masq and external-ip", opt)
```

Comment:
nit: serial comma?

Reply:
thanks! I did not know about the serial comma concept